<?xml version="1.0"?>
<launch>

    <remap from="joint_states"                          to="joint_states_1"/>
    <remap from="joint_command"                         to="joint_command_1"/>

    <remap from="shoulder_yaw_joint/command"            to="shoulder_yaw_joint_1/command"/>

    <remap from="shoulder_pitch_joint/command"          to="shoulder_pitch_joint_1/command"/>
    <remap from="shoulder_pitch_mimic_joint/command"    to="shoulder_pitch_mimic_joint_1/command"/>

    <remap from="elbow_pitch_joint/command"             to="elbow_pitch_joint_1/command"/>
    <remap from="elbow_pitch_mimic_joint/command"       to="elbow_pitch_mimic_joint_1/command"/>

    <remap from="wrist_pitch_joint/command"             to="wrist_pitch_joint_1/command"/>

    <remap from="wrist_roll_joint/command"              to="wrist_roll_joint_1/command"/>

    <remap from="gripper_revolute_joint/command"        to="gripper_revolute_joint_1/command"/>
    <remap from="gripper_prismatic_joint/command"       to="gripper_prismatic_joint_1/command"/>

    <!-- Load arbotix_driver -->
    <node name="arbotix_1" pkg="arbotix_python" type="arbotix_driver" output="screen">
        <!-- Load params in ROS parameter server -->
        <rosparam file="$(find phantomx_reactor_arm_controller)/config/arbotix_config_phantomx_wrist.yaml" command="load" />
    </node>

    <!-- This node coordinates the motors with the parallel/mimic joints -->
    <node name="phantomx_reactor_controller_1" pkg="phantomx_reactor_arm_controller" type="phantomx_reactor_parallel_motor_joints.py" output="screen"/>

    <!-- Start the node to publish and command the parallel gripper -->
    <node name="phantomx_reactor_parallel_gripper_1" pkg="phantomx_reactor_arm_controller" type="phantomx_reactor_gripper.py" output="screen" >
        <param name="rate" value="15"/>
    </node>

    <param name="arbotix_1/port" value="/dev/ttyUSB_REACTOR1"/>

</launch>